### PR TITLE
List subscribed topics when running topic list

### LINKED ISSUE #263: Topic list not listing subscribed topics
## Environment
* OS Version: Ubuntu 20.04
* Source or binary build?
Source both in Dome and Fortress branches

## Description
* Expected behavior: `ign topic -l` to list topics that are being subscribed to
* Actual behavior: Does not list topic until it's published then disappears

## Steps to reproduce

1. `ign gazebo velocity_control.sdf` (this also happens with `diff_drive.sdf` as well)
2. Add topic viewer in gui
3. `ign topic -e -t /model/vehicle_blue/cmd_vel`
4. `ign topic -t "/model/vehicle_blue/cmd_vel" -m ignition.msgs.Twist -p "linear: {x: 0.5}, angular: {z: 0.05}"`

## Output

I came across this when I wanted to check the green vehicle's initial `cmd_vel` and wasn't able to.

Peek 2021-09-24 12-53

### PR DESCRIPTION
**Pull request by caguero**
# 🎉 New feature

Requires https://github.com/gazebosim/gz-msgs/pull/322

Closes #263 

## Summary

This patch includes the subscribed topics in the output of `gz topic -l`.

## Test it

In one terminal run:

```
gz topic -e -t /foo
```

In another terminal run:

```
gz topic -l
```

Your output should be:
```
/foo
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by caguero** [comment]
@j-rivero , this PR requires `msgs10`. Is it possible to setup a `msgs10` nightly release to make CI happy?

File: include/gz/transport/NodeShared.hh [hunk 1]
```diff
@@ -218,6 +218,10 @@ namespace gz
       /// \param[in] _pub Information of the remote subscriber.
       public: void OnEndRegistration(const MessagePublisher &_pub);
 
+      /// \brief Callback executed when a SUBSCRIBERS request is received.
+      /// \param[in] _pub Request.
```
Comment (ahcorde) [inline include/gz/transport/NodeShared.hh:222]:
```suggestion
```

**Comment by Jaeyoung-Lim** [comment]
@caguero Thanks for adding this feature, this was a great pain point when trying to develop on gazebo

File: include/gz/transport/NodeShared.hh (hunk 1 above)
Comment (caguero) [reply include/gz/transport/NodeShared.hh]: Fixed in [a1af664](https://github.com/gazebosim/gz-transport/pull/379/commits/a1af664386900af32115be37d5ff0ccfce1af029)

**Comment by caguero** [comment]
> @j-rivero , this PR requires `msgs10`. Is it possible to setup a `msgs10` nightly release to make CI happy?



> @j-rivero , this PR requires `msgs10`. Is it possible to setup a `msgs10` nightly release to make CI happy?

Nightlies available: https://packages.osrfoundation.org/gazebo/ubuntu-nightly/pool/main/g/gz-msgs10/

**Comment by caguero** [comment]
CI is happy now, good to review!

**Review by ahcorde — APPROVED** [review]
LGTM, just a quick question: Is there any chance to backport this to Garden or lower versions ?

**Comment by caguero** [comment]
> LGTM, just a quick question: Is there any chance to backport this to Garden or lower versions ?

Not easily because we're breaking ABI here and in `gz-msgs`. That's why this is targeting `Harmonic`.

**Pull request merged**